sdks/python/src/opik/evaluation/models/base_model.py

```diff
@@ -1,21 +1,17 @@
 import abc
+from typing import Any
 
 
 class OpikBaseModel(abc.ABC):
     """
     This class serves as an interface to LLMs.
-
-    By the moment it's updated last time, str -> str relation
-    for generate and agenerate methods is enough, but it may
-    evolve in future if it becomes clear that we need models
-    not just consuming or returning strings only.
     """
 
     def __init__(self, model_name: str):
         self.model_name = model_name
 
     @abc.abstractmethod
-    def generate(self, input: str) -> str: ...
+    def generate(self, input: Any, **kwargs: Any) -> Any: ...
```

Comment:
Maybe it makes sense to rename `generate` into `generate_string` or something like that if we are keeping this method specifically as a simplified API when we just need a string output.
And to keep return type as "str".